### DISCUSSION TITLE
Motors not spinning equally

### DISCUSSION THREAD

**Discussion by it-global-architect** [iNavFlight/inav#10294] [Support]
Hi, I am building a VTOL and the behavior is strange, I will try to add as more information as possible.
It is my second aircraft using INAV and the first VTOL build, so you can expect that I'm doing something very stupid, but I have experience with fixed wings (I fly without fly controller) So, yes, I am noob in INAV and VTOL hahaha

1 - The main problem is about the 2 front motors in tricopter  mode, the left motor is spinning much stronger, some things that I tested and did not solved my problem:

- re calibrate the FC
- check the FC orientation. It is 270 degrees, I changed to default but it moved back to 270, I guess it is the default
- Turn on the airplane while holding it and check behavior. Besides the motor speed difference, everything else sounds ok related to orientation, if I roll the aircraft a lot, 45 degrees or more in the weaker motor direction, both will get a similar speed. 
- Compare real orientation with INAV Configurator status page. All sounds good. All directions sounds fine.
- Check stabilization reactions on surfaces, and yes, roll to one side raises the aileron on the other side and changes the motor speed. Nose up also makes the tail motor to spin stronger
- Checked the spin direction, I am using the reverse, spin out and it is configured as this in INAV

Besides the main problem, I also have another point and here the things will get crazy:

2 -Sometimes the aircraft, still landed, refuses to change from tricopter to airplane mode, the motor moves to transition point and back to tricopter mode. This is random and unplug and plug again the batteries will solve the problem after one or two reboots.

3 - Sometimes the tricopter flies like bouncing, left to right and front to back, looks like it is over correcting something

4 - The last and the most weird. One time I could take off successfully, (until now every single try was a fail), but once, just once, the aircraft was very stable, everything was really well, it was so well that I got some altitude and hit the transition, it stats well, so I hit the airplane mode, it worked but I was in angle mode and the aircraft was not responsive, could not make a turn easily and stated to fly away. So I set manual mode, the aircraft got extremely touchy (I fly huge 3D aircrafts and they are not touchy like this), but anyway, I set RTH before loose control and the aircraft was perfect, climb, came to me and start to fly around, I turned off RTH and landed it.  2 minutes later I tried to take off and it started to bounce like before.

I was using angle mode at this moment, but I did testes to take of tricopter in manual, horizon, acro and all failed as the aircraft was bouncing or sometimes rolling on the direction of the slower motor.

I record one video, it is shared at https://shorturl.at/0HkFv

P.S: I am not using magnetometer, I did not care about it because every single tutorial I was watching/reading suggested to turn it off if you primary fly in airplane mode, what is my case.

Some INAV configs I am using:
![image](https://github.com/user-attachments/assets/a74f4ca0-c0a2-4a21-b0bb-cf3ec11834df)

A strange thing is that INAV suggest motor 1 to be the tail one, but this will not work for vtol because in airplane mode it will try to use motors 1 and 2 only, so I set the tail as motor 3 and set the values as you can see bellow:
![image](https://github.com/user-attachments/assets/45683fbf-198e-4077-9a6c-68f5a6dd5101)

![image](https://github.com/user-attachments/assets/b53db541-1a04-4639-8988-1ea42d5ffbe3)
![image](https://github.com/user-attachments/assets/fdf1fe49-8880-4b02-8146-bffe27a1f486)
![image](https://github.com/user-attachments/assets/b10b185f-f71d-49d8-a350-a3dbbc629a84)
![image](https://github.com/user-attachments/assets/e80a6f10-f242-4f34-9826-1dede73aff08)

Diff all:
`Entering CLI Mode, type 'exit' to return, or 'help'

# 
# Building AutoComplete Cache ... Done!
# 
# status
INAV/MATEKF405SE_PINIO 7.1.2 May 31 2024 / 23:44:20 (4e1e59eb)
GCC-10.3.1 20210824 (release)
System Uptime: 5765 seconds
Current Time: 2041-06-28T01:04:00.000+00:00
Voltage: 0.00V (0S battery - NOT PRESENT)
CPU Clock=168MHz, GYRO=ICM42605, ACC=ICM42605, BARO=SPL06
STM32 system clocks:
  SYSCLK = 168 MHz
  HCLK   = 168 MHz
  PCLK1  = 42 MHz
  PCLK2  = 84 MHz
Sensor status: GYRO=OK, ACC=OK, MAG=NONE, BARO=OK, RANGEFINDER=NONE, OPFLOW=NONE, GPS=OK
SD card: Startup failed
Stack size: 6144, Stack address: 0x10010000, Heap available: 1528
I2C Errors: 0, config size: 10457, max available config: 131072
ADC channel usage:
   BATTERY : configured = ADC 1, used = ADC 1
      RSSI : configured = ADC 3, used = none
   CURRENT : configured = ADC 2, used = ADC 2
  AIRSPEED : configured = none, used = none
System load: 6, cycle time: 1010, PID rate: 990, RX rate: 9, System rate: 9
Arming disabled flags: NAV RX CLI
OSD: MSP DisplayPort: BetaFlight Compatability mode (HD) [53 x 20]
VTX: not detected
GPS: HW Version: UBLOX10 Proto: 34.09 Baud: 115200
  GNSS Capabilities:
    GNSS Provider active/default
    GPS 1/1
    Galileo 0/1
    BeiDou 1/1
    Glonass 1/0
    Max concurrent: 3

# diff all 

# version
# INAV/MATEKF405SE_PINIO 7.1.2 May 31 2024 / 23:44:20 (4e1e59eb)
# GCC-10.3.1 20210824 (release)

# start the command batch
batch start

# reset configuration to default settings
defaults noreboot

# resources

# Timer overrides
timer_output_mode 0 MOTORS

# Outputs [servo]
servo 1 1000 2000 1500 -100
servo 3 1000 2000 1500 -100
servo 4 1000 2000 1500 -100
servo 5 1000 2000 1500 -100

# safehome

# Fixed Wing Approach

# features
feature BAT_PROF_AUTOSWITCH
feature GPS
feature PWM_OUTPUT_ENABLE
feature FW_AUTOTRIM

# beeper

# blackbox
blackbox -NAV_ACC
blackbox NAV_POS
blackbox NAV_PID
blackbox MAG
blackbox ACC
blackbox ATTI
blackbox RC_DATA
blackbox RC_COMMAND
blackbox MOTORS
blackbox -GYRO_RAW
blackbox -PEAKS_R
blackbox -PEAKS_P
blackbox -PEAKS_Y

# Receiver: Channel map

# Ports
serial 2 33554432 115200 115200 0 115200
serial 4 2 115200 115200 0 115200

# LEDs

# LED color

# LED mode_color

# Modes [aux]
aux 0 0 0 1300 2100
aux 1 1 1 1300 1700
aux 2 12 1 1700 2100
aux 3 53 2 1300 1700
aux 4 11 2 1700 2100
aux 5 10 3 1300 2100
aux 6 21 4 1700 2100
aux 7 62 6 1300 2100
aux 8 63 6 1300 1700

# Adjustments [adjrange]

# Receiver rxrange

# temp_sensor

# Mission Control Waypoints [wp]
#wp 0 invalid

# OSD [osd_layout]
osd_layout 0 0 20 1 V
osd_layout 0 1 12 0 H
osd_layout 0 3 8 6 V
osd_layout 0 4 8 6 V
osd_layout 0 7 24 18 V
osd_layout 0 9 7 10 V
osd_layout 0 11 2 3 H
osd_layout 0 12 1 4 H
osd_layout 0 14 42 2 V
osd_layout 0 15 34 10 V
osd_layout 0 20 42 1 V
osd_layout 0 21 42 0 V
osd_layout 0 23 24 4 V
osd_layout 0 25 6 7 V
osd_layout 0 28 23 11 H
osd_layout 0 30 12 19 V
osd_layout 0 32 29 18 V
osd_layout 0 41 8 9 V
osd_layout 0 85 15 10 V
osd_layout 0 109 20 0 V
osd_layout 0 110 30 0 V
osd_layout 0 112 30 1 V

# Programming: logic
logic 0 0 0 0 0 0 0 0 0
logic 1 1 -1 6 1 10 0 0 0
logic 2 1 1 22 0 0 0 0 0

# Programming: global variables

# Programming: PID controllers

# OSD: custom elements

# master
set gyro_main_lpf_hz = 25
set dynamic_gyro_notch_q = 250
set dynamic_gyro_notch_min_hz = 30
set gyro_zero_y = -7
set gyro_zero_z = 3
set ins_gravity_cmss =  981.206
set acc_hardware = ICM42605
set acczero_x = -2
set acczero_y = -7
set acczero_z = 8
set accgain_x = 4095
set accgain_y = 4095
set accgain_z = 4098
set align_mag = CW270FLIP
set mag_hardware = NONE
set baro_hardware = SPL06
set serialrx_provider = CRSF
set motor_pwm_protocol = STANDARD
set failsafe_delay = 0
set failsafe_procedure = RTH
set reboot_character = 82
set small_angle = 180
set applied_defaults = 3
set gps_provider = UBLOX7
set gps_sbas_mode = AUTO
set gps_ublox_use_beidou = ON
set gps_ublox_use_glonass = ON
set airmode_type = STICK_CENTER_ONCE
set nav_wp_radius = 800
set nav_wp_max_safe_distance = 500
set nav_rth_allow_landing = FS_ONLY
set nav_rth_abort_threshold = 65000
set nav_rth_altitude = 5000
set nav_rth_home_altitude = 3000
set fw_loiter_direction = YAW
set nav_fw_control_smoothness = 2
set nav_fw_launch_motor_delay = 100
set nav_fw_launch_max_altitude = 5000
set nav_fw_launch_climb_angle = 25
set osd_video_system = BFHDCOMPAT

# mixer_profile
mixer_profile 1

set has_flaps = ON
set model_preview_type = 26
set motorstop_on_low = ON

# Mixer: motor mixer

mmix reset

mmix 0  1.000  0.000  0.000  0.100
mmix 1  1.000  0.000  0.000 -0.100

# Mixer: servo mixer
smix reset

smix 0 1 29 -100 0 -1
smix 1 2 29 -100 0 -1
smix 2 3 0 100 0 -1
smix 3 4 0 100 0 -1
smix 4 5 1 100 0 -1

# mixer_profile
mixer_profile 2

set motor_direction_inverted = ON
set platform_type = TRICOPTER
set model_preview_type = 1

# Mixer: motor mixer

mmix reset

mmix 0  1.000  1.000 -0.667  0.000
mmix 1  1.000 -1.000 -0.667  0.000
mmix 2  1.000  0.000  1.333  0.000

# Mixer: servo mixer
smix reset

smix 0 1 2 40 0 -1
smix 1 2 2 -40 0 -1
smix 2 1 29 60 0 -1
smix 3 2 29 60 0 -1
smix 4 3 0 100 0 -1
smix 5 4 0 100 0 -1
smix 6 5 1 100 0 -1
smix 7 1 38 -70 0 -1
smix 8 2 38 -70 0 -1

# profile
profile 1

set fw_p_pitch = 15
set fw_i_pitch = 5
set fw_d_pitch = 5
set fw_ff_pitch = 80
set fw_p_roll = 15
set fw_i_roll = 3
set fw_d_roll = 7
set fw_p_yaw = 50
set fw_i_yaw = 0
set fw_d_yaw = 20
set fw_ff_yaw = 255
set max_angle_inclination_rll = 450
set dterm_lpf_hz = 10
set fw_turn_assist_pitch_gain =  0.400
set nav_fw_pos_z_p = 25
set nav_fw_pos_z_d = 8
set nav_fw_pos_xy_p = 55
set d_boost_min =  1.000
set d_boost_max =  1.000
set rc_expo = 30
set rc_yaw_expo = 30
set roll_rate = 18
set pitch_rate = 9
set yaw_rate = 3

# profile
profile 2

# profile
profile 3

# battery_profile
battery_profile 1

# battery_profile
battery_profile 2

# battery_profile
battery_profile 3

# restore original profile selection
mixer_profile 2
profile 2
battery_profile 1

# save configuration
save`

Thanks so much!!!

**Comment by mmosca** [comment]
Does your esc support DSHOT?

If so, use DSHOT, otherwise, you need to calibrate your escs.

https://www.youtube.com/watch?v=1IrgbY0YhqM

**Reply by it-global-architect** [reply]
I did it and now looks like the motors are spinning correct but problem 3 remains. Anyway, already helped a lot, thanks.

**Comment by it-global-architect** [comment]
After change the motors to dshot looks like they are spinning correct but problem 3 remains. It is: 3 - Sometimes the tricopter flies like bouncing, left to right and front to back, looks like it is over correcting something

Something really weird happened: 
After change the config to dshot I went to fly it, the aircraft was bouncing and unstable, I was flying in tricopter mode using angle mode. So I changed the configuration on inav and replaced angle with horizon, I also did a PID reset at the same time. Save and fly. The aircraft was rock solid stable. I fly for around 5 minutes, only in tricopter mode because I was with camera unplugged and half battery only, but looks like problem solved. So I landed, charged the battery and plugged DJI O3 air unit and camera. Try to fly, and it was unstable and bouncing again. The only change at this time was the camera, so I unplugged it and try again, aircraft still bouncing. I have no idea anymore, the issue is too weird. I really appreciate any help. Thanks a lot.

**Comment by it-global-architect** [comment]
It already worked twice, only twice but already worked:
https://immich.ivolanski.com/share/42CX4GFY5YaHGgy7YrW90PnzKLFoX2uTRD6-on75xI-PY-ILsPJqmq_mFiFHMIOTI9w

Most of time it bounces like this:
https://immich.ivolanski.com/share/WPYB4C2tE6FkImr3Z-VDq2LcFxY2qjs23wRubCrt2CvlOBVQfbnoIVgqkRHshxgwO3E

And like this:
https://immich.ivolanski.com/share/1buKcqH9Bkm22a9hCWMBg-TQ4BlDfK5CnppT_5wMIJfLWomP6qQiTHmtTupZ4qsD_h0
